src/SonarScanner.MSBuild.TFS/IBuildVNextCoverageSearchFallback.cs

```diff
@@ -114,25 +127,32 @@ private void LogDebugFileList(string headerMessage, string[] files)
         }
 
         /// <summary>
-        /// Compares full file paths based on just the file name part
-        /// i.e. c:\aaa\bbb\file1.txt and c:\aaa\file1.txt are equal.
+        /// Compares file name and content hash tuples based on their hashes
         /// </summary>
-        internal class FileNameComparer : IEqualityComparer<string>
+        internal class FileHashComparer : IEqualityComparer<FileWithContentHash>
         {
-            public bool Equals(string x, string y)
+            public bool Equals(FileWithContentHash x, FileWithContentHash y)
             {
-                bool result = string.Equals(GetFileNameFromPath(x), GetFileNameFromPath(y), StringComparison.OrdinalIgnoreCase);
-                return result;
+                return string.Equals(x.ContentHash, y.ContentHash, StringComparison.OrdinalIgnoreCase);
```

Comment:
If you stick with the string approach, this should probably be `.Ordinal` instead of `.OrdinalIgnoreCase`